SVProgressHUD/SVProgressHUD.m

```diff
@@ -463,30 +436,37 @@ - (void)updateHUDFrame{
 }
 
 - (void)updateMask{
-    if(self.backgroundGradientLayer){
-        [self.backgroundGradientLayer removeFromSuperlayer];
-        self.backgroundGradientLayer = nil;
+    if(self.backgroundLayer){
+        [self.backgroundLayer removeFromSuperlayer];
+        self.backgroundLayer = nil;
     }
-    switch (self.maskType){
+    switch (self.defaultMaskType){
         case SVProgressHUDMaskTypeBlack:{
-            self.backgroundColor = [UIColor colorWithWhite:0 alpha:0.5];
+            
+            self.backgroundLayer = [CALayer layer];
+            self.backgroundLayer.frame = self.bounds;
+            self.backgroundLayer.backgroundColor = [UIColor colorWithWhite:0 alpha:0.5].CGColor;
+            CGPoint gradientCenter = self.center;
+            gradientCenter.y = (self.bounds.size.height - self.visibleKeyboardHeight) / 2;
+            [self.backgroundLayer setNeedsDisplay];
+            
+            [self.layer insertSublayer:self.backgroundLayer atIndex:0];
             break;
         }
             
         case SVProgressHUDMaskTypeGradient:{
-            self.backgroundColor = [UIColor clearColor];
-            self.backgroundGradientLayer = [SVRadialGradientLayer layer];
-            self.backgroundGradientLayer.frame = self.bounds;
+            SVRadialGradientLayer *layer = [SVRadialGradientLayer layer];
+            self.backgroundLayer = layer;
+            self.backgroundLayer.frame = self.bounds;
             CGPoint gradientCenter = self.center;
             gradientCenter.y = (self.bounds.size.height - self.visibleKeyboardHeight) / 2;
-            self.backgroundGradientLayer.gradientCenter = gradientCenter;
-            [self.backgroundGradientLayer setNeedsDisplay];
+            layer.gradientCenter = gradientCenter;
+            [self.backgroundLayer setNeedsDisplay];
             
-            [self.layer addSublayer:self.backgroundGradientLayer];
+            [self.layer insertSublayer:self.backgroundLayer atIndex:0];
```

Comment:
Add sublayer was originally used for the gradient view, which has a transparent center. When used with a solid color, the background layer would appear over the HUD. Insert at 0 ensures it will be the bottom-most layer.